src/Symfony/Component/Routing/RouteCollection.php

```diff
@@ -122,16 +122,17 @@ public function remove($name)
      * routes of the added collection.
      *
      * @param RouteCollection $collection A RouteCollection instance
+     * @param string $name
      *
      * @api
      */
-    public function addCollection(RouteCollection $collection)
+    public function addCollection(RouteCollection $collection, $name = '')
     {
         // we need to remove all routes with the same names first because just replacing them
         // would not place the new route at the end of the merged array
-        foreach ($collection->all() as $name => $route) {
-            unset($this->routes[$name]);
-            $this->routes[$name] = $route;
+        foreach ($collection->all() as $oldName => $route) {
+            unset($this->routes[$oldName]);
```

Comment:
it doesnt make sense to delete the old name, what you want is to delete the new name before adding it